drivers/platform/stm32/stm32_irq.h

```diff
@@ -0,0 +1,136 @@
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef IRQ_EXTRA_H_
+#define IRQ_EXTRA_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "irq.h"
+#include "stm32_hal.h"
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdbool.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @enum irq_id
+ * @brief Interrupts IDs supported by the STM32 irq driver
+ */
+enum irq_id {
+	/** External Interrupt ID1 */
+	EXTERNAL_INT_ID1,
+	/** External Interrupt ID2 */
+	EXTERNAL_INT_ID2,
+	/** External Interrupt ID3 */
+	EXTERNAL_INT_ID3,
+	/** External Interrupt ID4 */
+	EXTERNAL_INT_ID4,
+	/** External Interrupt ID5 */
+	EXTERNAL_INT_ID5,
+	/* Number of available interrupts */
+	NB_INTERRUPTS
+};
+
+/**
+ * @struct stm32_irq_init_param
+ * @brief Structure holding the STM32 Interrupt Event parameters.
+ */
+typedef struct stm32_irq_init_param {
+	/** GPIO Pin number */
+	uint8_t gpio_num;
+	/** IRQ type */
+	IRQn_Type irq_type;
+} stm32_irq_init_param;
+
+/**
+ * @struct stm32_irq_desc
+ * @brief Structure holding the STM32 Interrupt Event descriptor.
+ */
+typedef struct stm32_irq_desc {
+	/** GPIO Pin number */
+	uint8_t gpio_num;
+	/** IRQ type */
+	IRQn_Type irq_type;
+} stm32_irq_desc;
+
+/**
+ * @brief stm32 platform specific irq platform ops structure
+ */
+extern const struct irq_platform_ops stm32_irq_platform_ops;
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/* Initialize a interrupt controller peripheral. */
+int32_t stm32_irq_ctrl_init(struct irq_ctrl_desc **desc,
```

Comment:
All these functions could be deleted from .h since the functionality is exposed through ops.

Reply:
Make sense, I will delete them. However I need to move 'stm32_irq_platform_ops' structure variable defined in the stm32_irq.c file at the end to have visibility of the stm32 irq functions.